src/nodedumper.cc

```diff
@@ -39,24 +39,21 @@ void NodeDumper::handleIndent(const State &state)
 	including braces and indentation.
 	All children are assumed to be cached already.
  */
-std::string NodeDumper::dumpChildBlock(const AbstractNode &node)
+void NodeDumper::dumpChildBlock(const AbstractNode &node, std::stringstream &dump)
 {
-	std::stringstream dump;
 	if (!this->visitedchildren[node.index()].empty()) {
 		dump << " {\n";
-		const auto &chstr = dumpChildren(node);
-		if (!chstr.empty()) dump << chstr << "\n";
+		dumpChildren(node, dump);
 		dump << this->currindent << "}";
 	}
 	else {
 		dump << ";";
 	}
-	return dump.str();
 }
 
-std::string NodeDumper::dumpChildren(const AbstractNode &node)
+void NodeDumper::dumpChildren(const AbstractNode &node, std::stringstream &dump)
 {
-	std::stringstream dump;
+	bool empty = true;
 	for (auto child : this->visitedchildren[node.index()]) {
```

Comment:
We could optimize this a tiny step further and remove the need for that empty variable.
This will also make it possible to make this into a const function. Similar for `dumpChildBlock()`.

```
  const auto &it = this->visitedchildren.find(node.index());
  if (it == this->visitedchildren.end()) return;
```